Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "EditorGradientBakerComponent.h"
+
+#include <GradientSignal/Ebuses/GradientPreviewRequestBus.h>
+
+
+namespace GradientSignal
+{
+    void GradientBakerConfig::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serialize)
+        {
+            serialize->Class<GradientBakerConfig, AZ::ComponentConfig>()
+                ->Version(1)
+                ->Field("Gradient", &GradientBakerConfig::m_gradientSampler)
+                ->Field("InputBounds", &GradientBakerConfig::m_inputBounds)
+                ;
+
+            AZ::EditContext* edit = serialize->GetEditContext();
+            if (edit)
+            {
+                edit->Class<GradientBakerConfig>(
+                    "Gradient Baker", "")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::DisplayOrder, 50) // There's no special meaning to 50, we just need this class to move down and display below any children
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(0, &GradientBakerConfig::m_gradientSampler, "Gradient", "Input gradient to bake the output image from.")
+                    ->DataElement(0, &GradientBakerConfig::m_inputBounds, "Input Bounds", "Input bounds for where to sample the data.")
```

Comment:
Nit: Would be a bit more readable to use `AZ::Edit::UIHandlers::Default` instead of `0` for the first parameter on the DataElement.

Reply:
Will do!